elytron/WFLY-16306-keycloak-saml-feature-pack.adoc

```diff
@@ -0,0 +1,321 @@
+existing Keycloak project initially and then it will get moved to the new project with the rest of the adapter code.
+
+A new Galleon feature pack will be added to the Keycloak project with the group ID, `org.keycloak`, and artifact ID, `keycloak-saml-adapter-galleon-pack`.
+This new feature pack will contain two layers called `keycloak-saml` and `keycloak-client-saml`.
+
+When this feature pack and its layers are provisioned to a WildFly installation, the end result will mostly match the results
+of manually installing the Keycloak Elytron SAML adapter using the instructions linked above.
+
+In particular, when the `keycloak-saml` layer is provisioned, the following steps will occur:
+
+ # Install a set of modules under `modules/system/add-ons`
+ # Register the `org.keycloak.keycloak-saml-adapter-subsystem` extension.
+ # Add an empty `urn:jboss:domain:keycloak:1.4` subsystem.
+
+The `keycloak-client-saml` layer will depend on the `keycloak-saml` layer and when provisioned, the following additional steps will occur:
+
+ # Add the required resources to the `elytron` subsystem to result in an `http-authentication-factory` and `security-domain` supporting SAML being available.
+ # Update the `application-security-domain` mapping in the `undertow` subsystem to map the default security domain to the new `http-authentication-factory`.
+ # Update the `application-security-domain` mapping in the `ejb` subsystem  to use the new `security-domain`.
+
+The CLI scripts that would have been added via the manual installation steps won't be added to the resulting installation
+since the required resources will be added already during provisioning.
+
+Note that Keycloak and WildFly have different release cadences. This means anyone using Galleon to keep their WildFly
+installation up to date should be able to receive updates to the core WildFly installation independently of the new
+Keycloak feature pack. Having the new Keycloak feature pack installed should not prevent the server from being updated
+once later WildFly versions are available. Similarly, WildFly releases should not trigger the need for an update to Keycloak.
+
+==== Bare Metal vs. Cloud Use Cases
+
+For the bare metal use case, the `keycloak-client-saml` layer should be used.
+
+For WildFly s2i, when using the automatic registration of SAML deployments feature, the `keycloak-saml` layer should be used.
+(For this use case, the necessary `elytron`, `undertow`, and `ejb` subsystem configuration will be handled via a CLI script
+that is generated at boot time.)
+
+Otherwise, for WildFly s2i with manual deployments (i.e., with `keycloak-saml` configuration handled via a CLI script or included
+within the deployment itself), the `keycloak-client-saml` layer should be used.
+
+For bootable JAR on the cloud, the `keycloak-client-saml` layer should be used.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
```

Comment:
There are new environment variable defined in case of OIDC, see https://github.com/wildfly/wildfly-proposals/pull/434/files#diff-c82843e92d687d5f54d9f22aca4c6b079e0407cd653d4bb1fa58867951b3d779R90

Do we plan the corresponding step in the SAML case? I.e. using `SAML_` prefixed variables. If it's the case is it in the scope of the RFE or do we need another one?

Reply:
The OIDC_XXX env variables are defined by the cloud FP. No SAML_XXX env variables should be defined in the context of this RFE. Existing SSO_XXX env variable will be usable with cloud feature-pack + SAML feature-pack.